tket/src/serialize/pytket/encoder.rs

```diff
@@ -653,24 +641,28 @@ impl<H: HugrView> PytketEncoderContext<H> {
         }
 
         // Check that we have qubits or bits to attach the barrier command to.
-        //
-        // This should only fail when looking at the "leftover" unsupported nodes at the end of the decoding process.
         if op_values.qubits.is_empty() && op_values.bits.is_empty() {
-            return Err(PytketEncodeError::UnsupportedSubgraphHasNoRegisters {
+            // We cannot associate this subgraph to any qubit or bit register in the pytket circuit,
+            // so we'll store it in the [`AdditionalSubgraph`]s instead when finishing the encoding.
+            self.non_emitted_subgraphs.push(AdditionalSubgraph {
+                id: subgraph_id,
                 params: input_param_exprs.clone(),
             });
-        }
+        } else {
+            // If there's registers to attach the barrier command to, emit it as a regular command.
```

Comment:
```suggestion
            // If there are registers to which to attach, emit it as a barrier command.
```
("regular barrier command" if you like)